ui/src/app/workflows/components/workflows-list/workflows-list.tsx

```diff
@@ -126,7 +126,7 @@ export function WorkflowsList({match, location, history}: RouteComponentProps<an
         }
         storage.setItem('options', options, {} as WorkflowListRenderOptions);
 
-        const params = new URLSearchParams();
+        const params = new URLSearchParams(history.location.search);
```

Comment:
I found one side-effect of this change: if you visit http://localhost:8080/workflows?namespace=argo and refresh the page, it'll append `&namespace=argo` each time you refresh. That's happening because `historyUrl()` isn't deduplicating query parameters, which it should IMO. Here's a commit that fixes that: https://github.com/MasonM/argo-workflows/commit/8b48d5b627a17155c1a5a4b2778623d59aaa0e70

I haven't extensively tested that, so it's possible there's other side-effects